### DISCUSSION TITLE
How Can I make custom edge like default step type?

### DISCUSSION THREAD

**Discussion by Distrect** [xyflow/xyflow#3190] [Q&A] [answered]
I have very simple Custom Edge.

`import { BaseEdge, BezierEdge, getBezierPath } from "reactflow";

function CustomEdge(props) {
  const [edgePath, labelX, labelY] = getBezierPath({
    sourceX: props.sourceX,
    sourceY: props.sourceY,
    sourcePosition: props.sourcePosition,
    targetX: props.targetX,
    targetY: props.targetY,
    targetPosition: props.targetPosition,
  });

  return (
    // <BezierEdge className="react-flow__edge-step" {...props} />
     <BaseEdge
       path={edgePath}
       className="react-flow__edge-step"
       markerEnd={markerEnd}
     />
  );
}

export default CustomEdge;
`

I want this Custom Edge be like step edge type. I can use  `type:step` when connecting edge but i am trying to do here. Is that possible? Thank you for your answers

**Comment by bcakmakoglu** [comment, marked as answer]
React Flow provides a utility called `getSmoothStepPath`, just like `getBezierPath`, that you can use.

```tsx
import React from 'react';
import { BaseEdge, EdgeProps, getSmoothStepPath } from 'reactflow';

export default function CustomEdge({
  id,
  sourceX,
  sourceY,
  targetX,
  targetY,
  sourcePosition,
  targetPosition,
  style = {},
  markerEnd,
}: EdgeProps) {
  const [edgePath, labelX, labelY] = getSmoothStepPath({
    sourceX,
    sourceY,
    sourcePosition,
    targetX,
    targetY,
    targetPosition,
  });

  return <BaseEdge path={edgePath} markerEnd={markerEnd} style={style} />
}

**Reply by bcakmakoglu** [reply]
To get sharp instead of rounded corners for your smooth step edge (i.e. a step edge) just pass `borderRadius: 0` to `getSmoothStepPath`

```ts
  const [edgePath, labelX, labelY] = getSmoothStepPath({
    sourceX,
    sourceY,
    sourcePosition,
    targetX,
    targetY,
    targetPosition,
	borderRadius: 0,
  });
```

**Reply by Distrect** [reply]
Thank you very much sir. (Teşekkürler Kral)

**Comment by myescapekey** [comment]
How can I make it into first one 
<img width="510" alt="image" src="https://github.com/xyflow/xyflow/assets/161238722/e52ed162-b7c0-43c1-841e-8f4931e5a080">
<img width="405" alt="image" src="https://github.com/xyflow/xyflow/assets/161238722/c0b7f139-ae2e-4ac8-a963-d0c80d75dfce">
[tines.com](tines.com)
